sphereMove/SphereMove.cs

```diff
@@ -152,56 +200,115 @@ IEnumerator DelayedUpdate()
+                    break;
+                default:
+                    yield return new WaitForSeconds(updateInterval);
+                    break;
             }
+            
         }
 
-        // float cos = 1F, sin = 0F, scale = 0.1F;
-        // float trans_x = -0F, trans_y = -0F;
         // -------------- TODO ------------------
         // StopCoroutine("GetCarData") when race finished
         // Either stop it here in update() or stop it in the IEnumerator
     }
 
     // Return in format utcDateTime+durationSec
-    public static string GetNextSecond(DateTime utcDateTime, int durationSec = 10)
+    public static string GetNextSecond(string utcDateTime, int durationSec = 10)
+    {
+        //Debug.Log("Input " + utcDateTime);
+        // Parse the input string to DateTime object
+        if (DateTime.TryParseExact(utcDateTime, "yyyy-MM-ddTHH:mm:ss.ffffff",
+                                    CultureInfo.InvariantCulture,
+                                    DateTimeStyles.None,
+                                    out DateTime dt))
+        {
+            // Add one second to the parsed DateTime object
+            dt = dt.AddSeconds(durationSec);
+
+            // Format the DateTime object to ISO 8601 format
+            //Debug.Log("Output " + dt.ToString("yyyy-MM-ddTHH:mm:ss.ffffff"));
+            return dt.ToString("yyyy-MM-ddTHH:mm:ss.ffffff");
+        }
+        else
+        {
+            // Return an empty string if parsing fails
+            return string.Empty;
+        }
+    }
+
+    public static DateTime GetDateTime(string utcDateTime)
     {
         //Debug.Log("Input " + utcDateTime);
         // Parse the input string to DateTime object
-        //if (DateTime.TryParseExact(utcDateTime, "yyyy-MM-ddTHH:mm:ss.ffffff",
-        //                            CultureInfo.InvariantCulture,
-        //                            DateTimeStyles.None,
-        //                            out DateTime dt))
-        //{
-        // Add one second to the parsed DateTime object
-        DateTime dt = utcDateTime.AddSeconds(durationSec);
-
-        // Format the DateTime object to ISO 8601 format
-        //Debug.Log("Output " + dt.ToString("yyyy-MM-ddTHH:mm:ss.ffffff"));
-        return dt.ToString("yyyy-MM-ddTHH:mm:ss.ffffff");
-        //}
-        //else
-        //{
-        //    // Return an empty string if parsing fails
-        //    return string.Empty;
-        //}
+        if (DateTime.TryParseExact(utcDateTime, "yyyy-MM-ddTHH:mm:ss.ffffff",
+                                    CultureInfo.InvariantCulture,
+                                    DateTimeStyles.None,
+                                    out DateTime dt))
+        {
+            return dt;
+        }
+        else
+        {
+            Debug.Log("Cannot get DateTime!");
+            return DateTime.MinValue;
+        }
+    }
+
+    public enum State
+    {
+        Begin,
+        Running,
+        Ending,
+        Stopped
     }

```

Comment:
Perhaps we should comment out the `Ending` and `Stopped` state if they are currently not being called at all?

Reply:
It doesn't matter if we comment it or not, because I never turned the car into these two states. They are not currently affecting the code. But it is for sure something we need to do, as I somewhere wrote, we need to stop the car eventually. This logic is currently not implemented yet.